### PR TITLE
init normalization-sever

### PR DESCRIPTION
**Pull request by kingxueyuf**

### CODE REVIEW

**Comment by kingxueyuf** [comment]
PR for normalization-sever 

#TODO
Add logging
Add monitoring

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  <artifactId>normalization-sever</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <name>normalization-sever</name>
+
+  <properties>
+    <metamind.docker.image.base-haskell>
+      metamind-docker.jfrog.io/metamind/base-haskell:1.6.5
+    </metamind.docker.image.base-haskell>
+    <version.duckling>0.1.3.0</version.duckling>
```
Comment (imihov) [inline pom.xml]: Why define a property that is not used?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  </properties>
+
+  <dependencies>
+    <!-- Facebook/Duckling, for more details, please go to https://github.com/facebook/duckling -->
+    <dependency>
+      <groupId>facebook</groupId>
+      <artifactId>duckling</artifactId>
+      <version>0.1.3.0</version>
```
Comment (imihov) [inline pom.xml]: Use previously defined property

File: src/main/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
+FROM ${metamind.docker.image.base-haskell}
```
Comment (imihov) [inline src/main/docker/Dockerfile]: Unnecessary overuse of RUN command which should be avoided.

File: src/main/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+RUN mkdir /log
+
+RUN chown -R root /export/app/metamind/duckling
+
+WORKDIR /export/app/metamind/duckling
+
+RUN apt-get update && apt-get install -qq -y libpcre3 libpcre3-dev build-essential --fix-missing --no-install-recommends
+
+## Build duckling
+RUN stack upgrade
```
Comment (imihov) [inline src/main/docker/Dockerfile]: Makes the build irreproducible

File: src/main/docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+RUN stack setup
+
+RUN stack build
+
+## Excecute duckling binary
+ENTRYPOINT stack exec duckling-example-exe
```
Comment (imihov) [inline src/main/docker/Dockerfile]: What user does the service run as?

**Comment by vybs** [comment]
lgtm with the new changes.

**Comment by kingxueyuf** [comment]
Done
- Moved duckling build logic from normalization-server to docker-base/base-duckling
- Mount duckling log to host which follows Metamind log pattern

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                  <name>metamind</name>
+                  <alias>${project.name}</alias>
+                </network>
+                <ports>
+                  <port>8000:8000</port>
+                </ports>
+                <volumes>
+                  <bind>
+                    <volume>${project.basedir}/logs/application.log:/export/app/metamind/duckling/log/error.log</volume>
+                    <volume>${project.basedir}/logs/access.log:/export/app/metamind/duckling/log/access.log</volume>
```
Comment (kingxueyuf) [inline pom.xml]: mount
- /export/app/metamind/duckling/log/error.log to ${project.basedir}/logs/application.log
- /export/app/metamind/duckling/log/access.log to ${project.basedir}/logs/access.log

**Comment by ajmssc** [comment]
Can you add the logs folder to `.gitignore`

LGTM otherwise

**Comment by kingxueyuf** [comment]
Made changes based on @imihov 's instructions, will merge into master

**Pull request merged**